### PR TITLE
Add ROS 2 Documentation

### PR DESCRIPTION
**Pull request by rsanchez15**
Merge after:
-  https://github.com/eProsima/ros2_documentation/pull/4

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -455,3 +472,97 @@ def select_css(html_css_dir):
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 #
 # texinfo_no_detailmenu = False
+
+
+class RedirectFrom(Directive):
```
Comment (JLBuenoLopez-eProsima) [inline docs/conf.py:477]: Is this required or was it a try to include the ROS 2 links as they were without modifying them?

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: docs/rst/_templates/footer.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    {% endif %}
+    {% if 'ROS 2 Documentation' in title %}
+        <p>
+            © 2022 Open Robotics ROS 2 Documentation contributions included herein are the copyrights of their respective owners.
+            The documentation provided herein is licensed under the terms of the Creative Commons Attribution 4.0 International as published by Open Robotics.
+            ROS 2 and respective logos are trademarks of Open Robotics. All other trademarks are property of their respective owners.
+        </p>
+    {% endif %}
+    {{ super() }}
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/rst/_templates/footer.html:20]: Yes

File: docs/rst/_templates/footer.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{% extends "!footer.html" %}
+{% block extrafooter %}
+    {% if parents %}
+        {% if 'ROS 2 Documentation' in parents[0].title %}
+            <p>
+                © 2022 Open Robotics ROS 2 Documentation contributions included herein are the copyrights of their respective owners.
+                The documentation provided herein is licensed under the terms of the Creative Commons Attribution 4.0 International as published by Open Robotics.
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/rst/_templates/footer.html]: Already done in 925477d

File: docs/ros2_docs_spelling_wordlist.txt [hunk 4]
```diff
@@ -0,0 +1,772 @@
...
+XTypes
+Xu
+xUnit
+xwnb
+yaml
+Yau
+Yifei
+yml
+yuv
+zero
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/ros2_docs_spelling_wordlist.txt:1209]: Yes, every word in this file was extracted from the spelling error output.

File: docs/ros2_docs_spelling_wordlist.txt [hunk 5]
```diff
@@ -0,0 +1,772 @@
...
+Sture
+subdirectory
+subentities
+subfolder
+subitems
+submodule
+subprocess
+subscriptable
+subsubsections
+sucessfully
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/ros2_docs_spelling_wordlist.txt:1023]: Created task #15558 to keep track of this

File: docs/conf.py (hunk 1 above)
Comment (rsanchez15) [reply docs/conf.py:477]: This is required to not change every file that includes the `redirect-from` directive they have.

File: docs/conf.py [hunk 6]
```diff
@@ -111,20 +125,21 @@ def select_css(html_css_dir):
...
-    # spelling_word_list_filename = 'spelling_wordlist.txt'
     spelling_word_list_filename = [
         'rst/spelling_wordlist.txt',
+        'ros2_docs_spelling_wordlist.txt',
     ]
 
     from sphinxcontrib.spelling.filters import ContractionFilter
     spelling_filters = [ContractionFilter]
     spelling_ignore_contributor_names = False
+    # spelling_exclude_patterns = ['**/ros2_documentation/**']
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/conf.py]: Removed in b88f52e21f0f3e7af37f5626fb4309c659603801

File: docs/ros2_docs_spelling_wordlist.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply docs/ros2_docs_spelling_wordlist.txt:1209]: Maybe zero is not used in US English, though it sounds weird.

File: docs/ros2_docs_spelling_wordlist.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply docs/ros2_docs_spelling_wordlist.txt:1023]: Thanks!

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM! Thanks @rsanchez15

**Pull request merged**